lib/charms/mongodb/v0/shards_interface.py

```diff
@@ -53,27 +73,16 @@ def __init__(
         self.framework.observe(
             charm.on[self.relation_name].relation_changed, self._on_relation_event
         )
+        self.framework.observe(
+            charm.on[self.relation_name].relation_broken, self._on_relation_event
+        )
 
-        # TODO Follow up PR, handle rotating passwords
+        # TODO Future PR: handle self healing when all IP addresses of a shard changes and we have
+        # to manually update mongos
 
     def _on_relation_joined(self, event):
         """Handles providing shards with secrets and adding shards to the config server."""
-        if self.charm.is_role(Config.Role.REPLICATION):
-            self.charm.unit.status = BlockedStatus("role replication does not support sharding")
-            logger.error("sharding interface not supported with config role=replication")
-            return
-
-        if not self.charm.is_role(Config.Role.CONFIG_SERVER):
-            logger.info(
-                "skipping relation joined event ShardingRequirer is only be executed by config-server"
-            )
-            return
-
-        if not self.charm.unit.is_leader():
-            return
-
-        if not self.charm.db_initialised:
-            event.defer()
+        if not self.pass_hook_checks(event):
```

Comment:
Maybe some logging here could be useful for debug purposes?

Reply:
the command `pass_hook_checks` has [verbose logs for each scenario](https://github.com/canonical/mongodb-operator/blob/e8d7167762a262f6bcaf3a8c3c5460a46b720c86/lib/charms/mongodb/v0/shards_interface.py#L103-L127), we could include a log here but it wouldn't be as informative as the ones in `pass_hook_checks`.

I am happy to include a log message here. But not sure what I could say that would informative / not a restatement of what was logged in `pass_hook_checks`.

What would you like logged here? I can add it

Reply:
Yeah, nice logging there. I though of having something like "Skipping relation joined event: hook checks re not passed" or similar. But it is not a blocker I think 😄

Reply:
Perf, I'll add exactly that :) Thank you